wrappers/ortools.rb

```diff
@@ -407,6 +407,8 @@ def solve(vrp, job, thread_proc = nil, &block)
         }.uniq
         next if current_linked_ids.empty? && current_linked_vehicles.empty?
 
+        relation.lapse ||= 0 if relation.type == 'vehicle_trips'
```

Comment:
Protobuf already set nil values to zero for numerical fields

Reply:
In this case optimizer-ortools received -1. I will have to check what happens then.

Reply:
I think our code (ortools.rb) sends -1 by default.

Reply:
Could you think of a relation where having -1 or 0 would have different behavior ?

Reply:
I checked and no. The only difference would be when we check if lapse == -1 from tsp_simple side. I replaced it by checking lapse == 0 and in fact it even makes more sense because we don't want to generate a vehicle group duration relation where duration is 0.

https://github.com/Mapotempo/optimizer-ortools/pull/20

Reply:
Unless someone would like to "disable" a group of vehicles easily by setting their group duration 0. That would be the modification with least change and a lot easier then removing the vehicles or setting all their duration to 0. 

I wonder if that was the original reasoning of `-1`

Reply:
I am not sure because I think `-1` value is much older than `VehicleGroupDuration`. Also, those `VehicleGroupDuration` relations are not documented anymore. We are not sure about their performance. I am not even sure that providing group duration of 0 would allow to assign services on other vehicles...

Question is : do we want to allow customers to disable vehicles through this relation ? I would suggest to keep 0 as default relation.lapse value and add new relation (if needed one day) to disable a group of vehicles.

Reply:
This is not about allowing them or not using a valid option to do something valid. It is about expectations of the behavior of parameters.

If someone sets a duration of a vehicle (for example) to 1 second, it means 1 second of work. If they set it to 0, then it should mean 0 seconds of wotk. I don't think we should remove this possibility for no good reason. `-1` is clearly an "impossible" setting that we can use internally for the "turn off" logic. If we use 0 for this "turn-off" logic, we are using a valid setting which does the opposite of what it would normally do.

[principle of least surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) leads me to think that if a user sets the duration limit to `x`, then they would expect the vehicle to work no more than `x`. And it would be unexpected if x=0 would do the opposite. If the user didn't want a duration limit in the first place, they could just not supply the `duration` parameter or do `duration:  nil`

Reply:
I agree. In fact main problem is : why does ortools consider group_durations with no lapse (which used to correspond to lapse = -1). In preprocessing I will just filter relations that need a lapse and have no lapse (that would mean lapse is infinite, so relation is not needed). Like this if tsp_simple gets lapse = 0 it will do what we expect.